src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,219 @@
     }
+    Map<String, Object> traits = identify.traits();
+    for (Map.Entry<String, Object> entry : traits.entrySet()) {
+      String trait = entry.getKey();
+      Object value = entry.getValue();
+      trait = makeKey(trait);
+      String formattedValue;
+      if (value instanceof Date) {
+        Date dateValue = (Date) value;
+        formattedValue = FIREBASE_FORMAT.format(dateValue);
+      } else {
+        formattedValue = String.valueOf(value);
+      }
+      firebaseAnalytics.setUserProperty(trait, formattedValue);
+      logger.verbose("firebaseAnalytics.setUserProperty(%s, %s);", trait, formattedValue);
+    }
+  }
 
-    // mFirebaseAnalytics.setUserProperty(key, value);
+  @Override
+  public void track(TrackPayload track) {
+    super.track(track);
 
+    String event = track.event();
+    String eventName = makeKey(event);
+    Properties properties = track.properties();
+    Bundle formattedProperties = formatProperties(properties);
+    firebaseAnalytics.logEvent(eventName, formattedProperties);
+    logger.verbose("firebaseAnalytics.logEvent(%s, %s);", eventName, formattedProperties);
   }
 
-  @Override public void screen(ScreenPayload screen) {
-    super.screen(screen);
-//    mFirebaseAnalytics.setCurrentScreen(this, screen.name(), null /* class override */);
+  private Bundle formatProperties(Properties properties) {
+    Bundle bundle = new Bundle();
+    if ((properties.revenue() != 0 || properties.total() != 0)
+        && isNullOrEmpty(properties.currency())) {
+      bundle.putString(Param.CURRENCY, "USD");
+    }
+    for (Map.Entry<String, Object> entry : properties.entrySet()) {
+      Object value = entry.getValue();
+      String property = entry.getKey();
+      property = makeKey(property);
+      if (value instanceof Integer) {
+        int intValue = (int) value;
+        bundle.putInt(property, intValue);
+        logger.verbose("bundle.putInt(%s, %s);", property, intValue);
+        continue;
+      }
+      if (value instanceof Double) {
+        double doubleValue = (double) value;
+        bundle.putDouble(property, doubleValue);
+        logger.verbose("bundle.putDouble(%s, %s);", property, doubleValue);
+        continue;
+      }
+      if (value instanceof Long) {
+        long longValue = (long) value;
+        bundle.putLong(property, longValue);
+        logger.verbose("bundle.putLong(%s, %s);", property, longValue);
+        continue;
+      }
+      if (value instanceof String) {
+        String stringValue = String.valueOf(value);
+        bundle.putString(property, stringValue);
+        logger.verbose("bundle.putString(%s, %s);", property, stringValue);
+        continue;
+      }
+      if (value instanceof Date) {
+        Date dateValue = (Date) value;
+        String formattedDate = FIREBASE_FORMAT.format(dateValue);
+        bundle.putString(property, formattedDate);
+        logger.verbose("bundle.putString(%s, %s);", property, formattedDate);
+      }
```

Comment:
What if a propertye is not one of these types? Should the property be dropped completely? How about falling back to sending a string representation instead?

```
bundle.putString(property, String.valueOf(value));
logger.verbose("bundle.putString(%s, %s);", property, String.valueOf(value));
```